README.md

```diff
@@ -28,13 +27,298 @@ pip install -U pyinstaller-hooks-contrib
+
+
+#### Write tests(s)
+
+A test should be the least amount of code required to cause a breakage
+if you do not have the hook which you are contributing.
+For example if you are writing a hook for a library called `foo`
+which crashes immediately under PyInstaller on `import foo` then `import foo` is your test.
+If `import foo` works even without the hook then you will have to get a bit more creative.
+Good sources of such minimal tests are the hello world examples
+from the documentation of whichever library you're writing a hook for.
+Tests should not explicitly check for presence of data files or hidden modules directly -
+rather they should use parts of the library which are expected to use said data files or hidden modules.
+
+Tests currently all live in [src/_pyinstaller_hooks_contrib/tests/test_libraries.py](../master/src/_pyinstaller_hooks_contrib/tests/test_libraries.py).
+Navigate there and add something like the following, replacing all occurrences of `foo` with the real name of the library.
+(Note where you put it in that file doesn't matter.)
+
+```python
+@importorskip('foo')
+def test_foo(pyi_builder):
+    pyi_builder.test_source("""
+
+	# Your test here!
+        import foo
+
+	foo.something_fooey()
+
+    """)
+```
+
+If need be, you can specify version constraints using `@importorskip('foo', minversion="1.2.3")`
+if the library has changed significantly between versions.
+
+
+#### Run the test locally
+
+Running our full test suite is not recommended as it will spend a very long time testing code which you have not touched.
+Instead, run tests individually using either the `-k` option to search for test names:
+
+```
+pytest -k test_foo
+```
+
+Or using full paths:
+```
+pytest src/_pyinstaller_hooks_contrib/tests/test_libraries.py::test_foo
+```
+
+
+#### Run the test on CI/CD
+
+To test hooks on all platforms we use Github's continuous integration (CI/CD).
+Our CI/CD is a bit unusual in that it's triggered manually and takes arguments
+which limit which tests are run.
+This is for the same reason we filter tests when running locally -
+the full test suite runs like a dead snail on ketamine.
+
+First push the changes you've made so far.
+
+```commandline
+git push --set-upstream origin hook-for-foo
+```
+
+Replace *billy-the-buffalo* with your Github username in the following url then open it.
+It should take you to the `oneshot-test` actions workflow on your fork.
+You may be asked if you want to enable actions on your fork - say yes.
+```
+https://github.com/billy-the-buffalo/pyinstaller-hooks-contrib/actions/workflows/oneshot-test.yml
+```
+
+Find the **Run workflow** button and click on it.
+A dialog should appear containing one drop-down menu and 5 line-edit fields.
+This dialog is where you specify what to test and which platforms and Python versions to test on.
+Its fields are as follows:
+1.  A branch to run from. Set this to the branch which you are using (e.g. ``hook-for-foo``),
```

Comment:
Adding this to look at later - I believe the pyinstaller dev guide specifies to work on your fork's master branch, presumably if you're suggesting other branch names it's because this guide recommends branching somewhere.

Reply:
> I believe the pyinstaller dev guide specifies to work on your fork's master branch

If it does then it needs changing. Certainly I think the current maintainers all encourage creating branches. And yes, there's a recommendation to do so [at the top](https://github.com/pyinstaller/pyinstaller-hooks-contrib/pull/130/commits/5ac583f7e3f48ef142655ffd6c325a8020c81493#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R53-R58).

Reply:
Sorry, you're right. Hit send too quickly.